### ISSUE TITLE
Spectre error "Peer's certificate issuer has been marked as not trusted by the user"

### ISSUE THREAD

**Issue by muramasatheninja** [firefly-iii/firefly-iii#2148]
I am running Firefly III version 4.7.1.3

**Description**
After setting up spectre I'm not able to get it to connect. I get the following error

[2019-03-05 02:50:17] local.ERROR: Exception is: {"class":"FireflyIII\\Exceptions\\FireflyException","errorMessage":"Request Exception: cURL error 60: Peer's certificate issuer has been marked as not trusted by the user.","time":"Tue, 05 Mar 2019 02:50:17 -0700","file":"\/usr\/share\/nginx\/html\/firefly-iii\/app\/Services\/Spectre\/Request\/SpectreRequest.php","line":245,"code":0,"version":"4.7.1.3"}  
[2019-03-05 02:50:18] local.ERROR: Request Exception: cURL error 60: Peer's certificate issuer has been marked as not trusted by the user. {"userId":1,"email":"myuseremail@example.com","exception":"[object] (FireflyIII\\Exceptions\\FireflyException(code: 0): Request Exception: cURL error 60: Peer's certificate issuer has been marked as not trusted by the user. at /usr/share/nginx/html/firefly-iii/app/Services/Spectre/Request/SpectreRequest.php:245)"} 

"Peer's certificate issuer has been marked as not trusted by the user" seems to be the gist of the problem. Unfortunately I have no idea what this is trying to tell me or how to fix it. Any help appreciated!

**Comment by JC5** [comment]
This happens when the trust store on the local machine is particularly old. It depends on your system how you can update it.

**Comment by muramasatheninja** [comment]
I'm running it on a CentOS 7 Container on an Ubuntu LXD host. Hopefully it is as simple a just updating some things.

**Comment by JC5** [comment]
OK, tell me if there is a file in this location: `/usr/local/ssl/cert.pem`, and what is the file size?

**Comment by muramasatheninja** [comment]
Ok so I don't have a /usr/local/ssl directory. I do have /etc/ssl/certs though

`lrwxrwxrwx 1 root root   49 Mar  5 10:33 ca-bundle.crt -> /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
lrwxrwxrwx 1 root root   55 Mar  5 10:33 ca-bundle.trust.crt -> /etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt`

/etc/pki/ca-trust/extracted/pem
`-rw-r--r-- 1 root root    898 May 14  2018 README
-r--r--r-- 1 root root 177221 Mar  5 10:35 email-ca-bundle.pem
-r--r--r-- 1 root root      0 Mar  5 10:35 objsign-ca-bundle.pem
-r--r--r-- 1 root root 211658 Mar  5 10:35 tls-ca-bundle.pem
`

**Comment by JC5** [comment]
OK, does the `tls-ca-bundle.pem` file have entries like this one?

```
GlobalSign Root CA
==================
-----BEGIN CERTIFICATE-----
(bunch of random text)

GlobalSign Root CA - R2
=======================
-----BEGIN CERTIFICATE-----
(more random text)

```

**Comment by muramasatheninja** [comment]
Yes it does. Should I add the entry given to me in the setup step from firefly?

**Comment by JC5** [comment]
You should back up the file and replace the content with that of Firefly III’s file. Then reboot the machine.

**Comment by muramasatheninja** [comment]
Ok so I tried copying the cert located at /vendor/rmccue/requests/library/Requests/Transport/cacert.pem and then rebooting the machine and it didn't seem to have any effect. Am I working with the right file? I'm guessing that if I can get the machine to trust it then I will hopefully be good to go.

**Comment by JC5** [comment]
Try the file in the .deploy directory.

**Comment by muramasatheninja** [comment]
Ok so I took firefly-iii/.deploy/docker/cacert.pem. Dropped it into '/etc/pki/ca-trust/source/' and run "update-ca-trust extract" and it moves it into that folder. However it still doesn't seem to recognize it.

**Comment by JC5** [comment]
Strange. I'm sorry, there's not much I can do at this point.

What happens when you do `curl www.saltedge.com`?

**Comment by muramasatheninja** [comment]
curl saltedge returns
`<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
`
No worries, I'll keep playing with it. It's possible that something screwy is going on with my container. I'll play with it somemore and report back.

**Comment by JC5** [comment]
OK, so that's a bit weird. I remember a similar issue in the past, let me look for it. It had to do with the root certificates though.

**Comment by JC5** [comment]
Check out this issue: https://github.com/firefly-iii/firefly-iii/issues/1402

**Comment by muramasatheninja** [comment]
Ok so that was helpful for me finding out what the problem was. Which was mostly me not paying attention.

I followed the instructions for setting up with this guide. https://old.reddit.com/r/FireflyIII/comments/825n4l/centos_7_nginx_installation_guide/

But I didn't notice that the version specified 4.7.1.3 the latest being 4.7.15. After upgrading to 4.7.15 I was able to connect to spectre. Now the only down side is that my bank is not supported. But this issue is resolved.

Thanks for all the help
